tools/driver/autolink_extract_main.cpp

```diff
@@ -107,26 +107,38 @@ class AutolinkExtractInvocation {
 
 /// Look inside the object file 'ObjectFile' and append any linker flags found in
 /// its ".swift1_autolink_entries" section to 'LinkerFlags'.
-static void
+/// Return 'true' if there was an error, and 'false' otherwise.
+static bool
 extractLinkerFlagsFromObjectFile(const llvm::object::ObjectFile *ObjectFile,
-                                 std::vector<std::string> &LinkerFlags) {
+                                 std::vector<std::string> &LinkerFlags,
+                                 CompilerInstance &Instance) {
   // Search for the section we hold autolink entries in
   for (auto &Section : ObjectFile->sections()) {
     llvm::StringRef SectionName;
     Section.getName(SectionName);
     if (SectionName == ".swift1_autolink_entries") {
-      llvm::StringRef SectionData;
-      Section.getContents(SectionData);
+      llvm::Expected<llvm::StringRef> SectionData = Section.getContents();
+      if (!SectionData) {
+        std::string message;
+        {
+          llvm::raw_string_ostream os(message);
+          logAllUnhandledErrors(SectionData.takeError(), os, "");
+        }
+        Instance.getDiags().diagnose(SourceLoc(), diag::error_open_input_file,
+                                  ObjectFile->getFileName() , message);
+        return true;
+      }
 
       // entries are null-terminated, so extract them and push them into
       // the set.
       llvm::SmallVector<llvm::StringRef, 4> SplitFlags;
-      SectionData.split(SplitFlags, llvm::StringRef("\0", 1), -1,
-                        /*KeepEmpty=*/false);
+      SectionData->split(SplitFlags, llvm::StringRef("\0", 1), -1,
+          /*KeepEmpty=*/false);
```

Comment:
The original indentation here looks fine. I don't think you should change it (except to add a space to keep  it aligned with the open paren).